deeppavlov/core/agent_v2/agent.py

```diff
@@ -38,4 +36,13 @@ def __call__(self, utterances: Sequence[str], user_telegram_ids: Sequence[Hashab
         self.state_manager.add_bot_utterances(me_dialogs, utterances, [datetime.utcnow()] * len(me_dialogs),
                                               skill_names, confidences)
 
+        self._update_annotations(me_dialogs)
+
         return utterances  # return text only to the users
+
+    def _update_annotations(self, me_dialogs: Sequence[Dialog]):
```

Comment:
Let's move this functionality to StateManager. Only StateManager should be responsible for modifying and saving me objects.